### ISSUE TITLE
Static ipv4 gets ignored/reset by ipv6 (Win7 Host, CentOS7 Guests)

### ISSUE THREAD

**Issue by lucabelluccini** [mitchellh/vagrant#7791]
### Virtualbox Version
stable, 5.0.26

### Vagrant version
Vagrant 1.8.5

### Host operating system
Windows 7 

### Guest operating system
Several boxes using bento/centos-7.1

### Vagrantfile
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

VAGRANTFILE_API_VERSION = "2"
ENVIRONMENT_NAME = "dev"
TMP_FOLDER = "/tmp"
DOMAIN = "escluster"
PUPPETMASTER_HOSTNAME = "puppetmaster"
PUPPETMASTER_IP = "172.16.0.253"
NODE_IP_BASE = "172.16.0"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|

  # Read data from xml 
  require 'rexml/document'
  include REXML
  
  # Open file
  xmlfile = File.new("cluster.xml")
  xmldoc = Document.new(xmlfile)
  
  # Read domain
  network = XPath.match(xmldoc, "/cluster/network")
  domain = network[0].attributes["domain"]
  
  # Read machines hostnames and generate ips + puppetmaster
  @machines = []
  @machines.push(hostname: PUPPETMASTER_HOSTNAME, ip: PUPPETMASTER_IP, full_hostname: "#{PUPPETMASTER_HOSTNAME}.#{domain}")
  counter = 21
  xmldoc.elements.each("/cluster/machines/machine") do |e|
  	h = e.attributes["hostname"]
  	@machines.push(hostname: h, ip: "#{NODE_IP_BASE}.#{counter}", full_hostname: "#{h}.#{domain}")
  	counter += 1
  end
  
  puts @machines
  
  # Generate hosts file update script
  hosts_script = "#!/bin/sh\n\n"
  @machines.each do |m|
  	hosts_script << 'echo "' << m[:ip] << "\t" << m[:full_hostname] << "\t" << m[:hostname] << '" | sudo tee -a /etc/hosts' << "\n"
  end
  hosts_script << 'cat /etc/hosts'
  puts hosts_script

  config.vm.box = "bento/centos-7.1" # "centos/7"

  # FIXME the keys should work but... not for me
  config.ssh.username = 'vagrant'
  config.ssh.password = 'vagrant'
  config.ssh.insert_key = 'false'
  
  config.vm.synced_folder '.', '/vagrant', disabled: true
  
  puppetm = @machines[0]
  config.vm.define "puppetmaster" do |puppetmaster|
	puppetmaster.vm.hostname = puppetm[:full_hostname]
	puppetmaster.vm.network "private_network", ip: puppetm[:ip]
	puppetmaster.vm.provision "shell", run: "always", privileged: true, inline: hosts_script
  end

  @machines.drop(1).each do |machine|
	full_hostname = machine[:full_hostname]
	ip = machine[:ip]
    config.vm.define full_hostname do |node|
      node.vm.hostname = full_hostname
	  node.vm.network "private_network", ip: ip
	  node.vm.provision "shell", run: "always", privileged: true, inline: hosts_script
    end
  end
  
end

```

cluster.xml file
```
<cluster>
    <puppet environment="dev"/>
	<network domain="escluster"/>
	<machines>
		<machine hostname="esnode1" role="elasticsearch"/>
		<machine hostname="esnode2" role="elasticsearch"/>
		<machine hostname="esnode3" role="elasticsearch"/>
	</machines>
</cluster>
```

Please note, if you are using Homestead or a different Vagrantfile format, we
may be unable to assist with your issue. Try to reproduce the issue using a
vanilla Vagrantfile first.

### Debug output
https://gist.github.com/lucabelluccini/1bbe744a81e636508b430abe3e9a11ba

### Expected behavior
Each VM should have its own static ip

### Actual behavior
Each VM can eventually get its own ipv4 static ip after a network service restart or randomly during the time the interface gets an ipv6 (and ipv4 gets removed).

```
[vagrant@esnode1 ~]$ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: enp0s3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 08:00:27:f6:b0:07 brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic enp0s3
       valid_lft 86396sec preferred_lft 86396sec
3: enp0s8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 08:00:27:a9:38:79 brd ff:ff:ff:ff:ff:ff
    inet 172.16.0.21/24 brd 172.16.0.255 scope global enp0s8
       valid_lft forever preferred_lft forever
[vagrant@esnode1 ~]$ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: enp0s3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 08:00:27:f6:b0:07 brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic enp0s3
       valid_lft 86355sec preferred_lft 86355sec
3: enp0s8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 08:00:27:a9:38:79 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::a00:27ff:fea9:3879/64 scope link
       valid_lft forever preferred_lft forever

```

### Steps to reproduce
1. Use above vagrant file and cluster.xml
2. Execute vagrant up
3. Access to the machines... you can find the correct ip but after few seconds it gets overridden

### References
Found similar issues but they are all closed

**Comment by lucabelluccini** [comment]
Update:
bento/centos-7.2
ubuntu/trusty64

Are working correctly. I would like to troubleshoot this but I cannot identify the process updating the network interfaces. Any idea?

**Comment by chrisroberts** [comment]
@lucabelluccini To confirm I understand correctly. `bento/centos-7.1` is working _incorrectly_ and `bento/centos-7.2` is working _correctly_ ?

**Comment by lucabelluccini** [comment]
Exactly, the 7.2 works correctly, NOT overriding my ipv4 static ip.

**Comment by chrisroberts** [comment]
Fixed by: #7751

**Comment by chrisroberts** [comment]
Fixed via #7751
